File: initial/src/main/java/com/example/accessingdatamysql/Address.java

package com.example.accessingdatamysql;

import javax.jws.soap.SOAPBinding;
import javax.persistence.*;

@Entity // This tells Hibernate to make a table out of this class
@Table(name = "address")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer addressID;
/*    public PostalCodeCity getPostalCodeCity() {
        return postalCodeCity;
    }

    public void setPostalCodeCity(PostalCodeCity postalCodeCity) {
        this.postalCodeCity = postalCodeCity;
    }*/




/*@OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "addressID")
    private User user;*//*


    */
/*@OneToOne
    @JoinColumn(name = "city")
    private PostalCodeCity postalCodeCity;*//*

*/

    private String addressLine;

    @OneToOne
    @JoinColumn(name = "postalCodeCity")
    private PostalCodeCity postalCodeCity;

    public PostalCodeCity getPostalCodeCity() {
        return postalCodeCity;
    }

    public void setPostalCodeCity(PostalCodeCity postalCodeCity) {
        this.postalCodeCity = postalCodeCity;
    }

    public Address() {
    }

    public Address(String postalCode, String addressLine) {
        this.addressLine = addressLine;
    }

/*
    @OneToOne(mappedBy = "sellerAddressID")
    public User user;

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
*/

    public Integer getAddressID() {
        return addressID;
    }

    public void setAddressID(Integer addressID) {
        this.addressID = addressID;
    }

    public String getAddressLine() {
        return addressLine;
    }

    public void setAddressLine(String addressLine) {
        this.addressLine = addressLine;
    }

}

File: lib/models/address.dart
class Address {
  int addressID;
  String addressName;
  String addressLine;
  String postalCode;
  String city;
  String country;
  Address(
      {this.addressID,
      this.addressName,
      this.addressLine,
      this.postalCode,
      this.city,
      this.country});
  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      addressID: json['addressID'],
      addressName: json['addressName'],
      addressLine: json['addressLine'],
      postalCode: json['postalCode']['postalCode'],
      city: json['postalCode']['city'],
      country: json['postalCode']['country'],
    );
  }
  Map<String, dynamic> toJson() {
    return {
      "addressLine": this.addressLine,
      "addressName": this.addressName,
      "postalCode": {
        "postalCode": this.postalCode,
        "city": this.city,
        "country": this.country
      }
    };
  }
}
